src/MeshKernel/CurvilinearGrid.cpp

```diff
@@ -136,7 +111,29 @@ std::tuple<std::vector<meshkernel::Point>, std::vector<meshkernel::Edge>, std::v
     return {nodes, edges, gridIndices};
 }
 
-std::tuple<int, int> meshkernel::CurvilinearGrid::GetNodeIndices(Point point)
+bool meshkernel::CurvilinearGrid::IsValid() const
+{
+    if (m_gridNodes.empty())
+    {
+        return false;
+    }
+    if (m_gridNodes[0].empty())
+    {
+        return false;
+    }
+    if (m_gridNodes.size() < 2)
+    {
+        return false;
+    }
+    if (m_gridNodes[0].size() < 2)
+    {
+        return false;
+    }
+
+    return true;
+}
+
+meshkernel::CurvilinearGrid::NodeIndices meshkernel::CurvilinearGrid::GetNodeIndices(Point point)
```

Comment:
you have here (and in other locations) really long typenames such as meshkernel::CurvilinearGrid::NodeIndices
You could make them smaller by using a "typedef" 
or alternatively put "using namespace meshkernel::CurvilinearGrid" at the top

Reply:
thanks, accounted for in the next commit